### PR TITLE
Adding pyvpt2 feedstock

### PR DESCRIPTION
**Pull request by philipmnel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pyvpt2/meta.yaml
Comment (loriab) [inline recipes/pyvpt2/meta.yaml:2]: I'd start with 0.1, as some machinery might only expect 0.0.* as a placeholder for on-the-fly computed versions

File: recipes/pyvpt2/meta.yaml
Comment (loriab) [inline recipes/pyvpt2/meta.yaml:26]: I think with that `from pydantic.v1 import ...` you don't need this restriction

File: recipes/pyvpt2/meta.yaml
Comment (loriab) [inline recipes/pyvpt2/meta.yaml:36]: if there's any quick script akin to https://github.com/conda-forge/libefp-feedstock/blob/main/recipe/test_pylibefp.py , https://github.com/conda-forge/libefp-feedstock/blob/main/recipe/meta.yaml#L122 , https://github.com/conda-forge/libefp-feedstock/blob/main/recipe/meta.yaml#L135 you can set up to run, it'll make the recipe more robust.

File: recipes/pyvpt2/meta.yaml
Comment (loriab) [inline recipes/pyvpt2/meta.yaml:37]: I've also found a line like https://github.com/conda-forge/qcfractal-feedstock/blob/main/recipe/meta.yaml#L69 helpful to make sure the version gets computed right for the conda package.

**Review by loriab — COMMENTED** [review]
otherwise, lgtm!

File: recipes/pyvpt2/meta.yaml
Comment (philipmnel) [reply recipes/pyvpt2/meta.yaml:36]: I'm not sure that I have a quick test that could be run, but I did add the version checking.

**Comment by philipmnel** [comment]
@conda-forge/help-python Is there anything blocking this PR?

**Comment by philipmnel** [comment]
@conda-forge/help-python This is ready for review from a maintainer. Thanks!

**Review by xhochy — APPROVED** [review]

**Pull request merged**
